### PR TITLE
Thrift codegen improvements

### PR DESCRIPTION
**Pull request by mahwy**
- Removed ThriftServiceHelper
- An easy way to create an underlying thrift service(ReqRepServicePerEndpoint)
- Options to customize: trace data serializer, span name prefix  and tracing service

### CODE REVIEW

File: projectile-export/src/main/scala/com/kyleu/projectile/models/feature/core/thrift/ThriftServiceFile.scala [hunk 1]
```diff
@@ -10,28 +10,85 @@ import com.kyleu.projectile.models.thrift.input.{ThriftFileHelper, ThriftMethodH
...
+         |* {{{
+         |* val svcPerEndpoint = $classCanonicalName.mkServicePerEndpoint(url)
+         |* val myService = $classCanonicalName
+         |*   .from(svcPerEndpoint)
+         |*   .withTracingService(openTracingTracingService)
+         |*   .withThriftSpanNamePrefix("myThriftService")
+         |*   .withTraceDataSerializer(td => Map.empty)
+         |* }}}
+         |*/
+       """.stripMargin)
```
Comment (KyleU) [inline projectile-export/src/main/scala/com/kyleu/projectile/models/feature/core/thrift/ThriftServiceFile.scala]: Doesn't this screw with the indention?

Comment (mahwy) [reply projectile-export/src/main/scala/com/kyleu/projectile/models/feature/core/thrift/ThriftServiceFile.scala]: It works fine, at least for the javadoc comment.

**Comment by mahwy** [comment]
@KyleU I did more code clean up. Please review.

File: projectile-export/src/main/scala/com/kyleu/projectile/models/feature/core/thrift/ThriftServiceFile.scala (hunk 1 above)
Comment (KyleU) [reply projectile-export/src/main/scala/com/kyleu/projectile/models/feature/core/thrift/ThriftServiceFile.scala]: The auto-formatter will rewrite it, which causes every compile to rewrite files twice, once for codegen, once formatted. I'll merge it, fix the formatting, and publish a new version. Thanks for the contribution

**Pull request merged**